perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -3183,12 +3187,28 @@ private void processMemberAfterRemovingLastSponsor(PerunSession sess, Member spo
 
 			// we must pass member back for the purpose of validation
 			status.put(MEMBER, member);
-
 			status.put(STATUS, OK);
 		} catch (Exception e) {
 			log.error("Failed to create a sponsored user.", e);
 			status.put(STATUS, e.getMessage());
 		}
+
+		if (groups != null && !groups.isEmpty()) {
+			Map<Integer, String> groupAssignmentErrors = new HashMap<>();
+			if (member != null) {
+				for (Group group : groups) {
+					try {
+						perunBl.getGroupsManagerBl().addMember(sess, groups, member);
```

Comment:
You should pass `group` and not `groups`.

Reply:
Fixed.